### DISCUSSION TITLE
How to use lemminx-maven extension in Neovim

### DISCUSSION THREAD

**Discussion by LorenzoBettini** [eclipse/lemminx-maven#588] [Q&A]
Using lemminx LSP in Neovim is easy because it is supported by Mason.

However, it is not clear to me, and I can't find further information, how to use this extension when using the lemminx LSP in Neovim.

Thanks in advance
Lorenzo

**Comment by mickaelistria** [comment]
In the README: "Then, just start the LemMinX Language Server as documented on the LemMinX documentation, and make sure the LemMinX-Maven extension is in the classpath. "

**Reply by LorenzoBettini** [reply]
But is this extension available as a jar somewhere?

**Reply by mickaelistria** [reply]
In the same README, there is https://github.com/eclipse/lemminx-maven?tab=readme-ov-file#consuming-lemminx-maven

**Reply by angelozerr** [reply]
@LorenzoBettini I think this PR https://github.com/eclipse/lemminx-maven/pull/434 should interest you

**Reply by LorenzoBettini** [reply]
I'll have a look at the PR.
However, @mickaelistria , the lemminx downloaded by Mason is the one from Microsoft, which I seem to understand is a binary so the classpath cannot be passed, unless I miss something.

**Reply by mickaelistria** [reply]
Sorry, I cannot help further on that aspect, my focus and knowledge here is limited to official LemMinX artifact, which are jar files distributed though Eclipse Nexus instance.
It's possible that you can find somewhere some binary builds of LemMinX compiled with Graal or Quarkus or whatever similar technologies, but this is not fully supported and I doubt that extensibility can work with those.

**Reply by angelozerr** [reply]
I confirm that if you want to consume lemminx maven you cannot use lemminx binary built with graalvm.

We did that in vscode-xml because a lot of users didnt want to install java runtime.

But in vscode xml it is possible to start lemminx with the Java command and even set lemminx extension with user settings to load some extension like lemminx maven.

If you want to consume lemminx maven in neovim you will need to start lemminx with Java command and add lemminx maven in the classpath

**Comment by sergii-dudar** [comment]
I don't know how actual question is, but in case someone will try to make it works in neovim - my solution:

- installation and preparing lemminx ls jar with maven extension: https://github.com/sergii-dudar/dotfiles/blob/master/bin/install/nvim/ls/lemminx-maven/load_and_build_all.sh (can be called for example by 'build' lazy.vim method, or maybe even provide installation option MR to mason registry, but for me completely fine to do it once manually)

- my nvim lsp config (I'm usin LazyVim based config & blink.cmp) https://github.com/sergii-dudar/dotfiles/blob/master/nvim/.config/nvim/lua/plugins/editor/java/lemminx-ls.lua

lsp complete, go to definition and related stuff working perfectly! Thanks to answers that provided to me understanding how it should be properly configured in the end ❤️

<img width="2178" height="2099" alt="image" src="https://github.com/user-attachments/assets/c8022d3c-dc53-4e53-bf67-d9d7a6e2c7c7" />

<img width="1838" height="998" alt="image" src="https://github.com/user-attachments/assets/e2feefef-4b7b-4b14-83bf-0c8fca9eca47" />

<img width="2157" height="1715" alt="image" src="https://github.com/user-attachments/assets/a42925a8-ed82-4560-b923-958f4f8ff22f" />

**Comment by angelozerr** [comment]
I wonder if you should do like consumming lemminx-maven in vscode, see https://github.com/eclipse-lemminx/lemminx-maven/blob/cb7fc9cbe7353b0ab0122caa4e7cdd5b5b6e3f41/CONTRIBUTING.md#building-and-running-lemminx-maven-in-vscode-xml-extension

@vrubezhny what do you think about that?
